monai/transforms/intensity/array.py

```diff
@@ -242,10 +244,16 @@ def __call__(self, img: NdarrayOrTensor, offset: float | None = None) -> Ndarray
 
         img = convert_to_tensor(img, track_meta=get_track_meta())
         offset = self.offset if offset is None else offset
-        out = img + offset
-        out, *_ = convert_data_type(data=out, dtype=img.dtype, safe=self.safe)
-
-        return out
+        if self.channel_wise:
+            for i, d in enumerate(img):
+                out = d + offset
```

Comment:
Thanks, but if you don't randomize offset here, the output won't make any difference since offset is a constant.

Reply:
Ahh yes good catch, how do you propose I do this ?

Reply:
Hi @SauravMaheshkar, I would suggest you refer to this [PR](https://github.com/Project-MONAI/MONAI/pull/6793).
Thanks!